crates/oxc_linter/src/rules/eslint/no_unused_vars/symbol.rs

```diff
@@ -186,8 +186,11 @@ impl<'a> Symbol<'_, 'a> {
     fn in_export_node(&self) -> bool {
         for parent in self.nodes().ancestors(self.declaration_id()).skip(1) {
             match parent.kind() {
-                AstKind::ModuleDeclaration(module) => {
-                    return module.is_export();
+                m if m.is_module_declaration() => {
+                    let module = m.as_module_declaration_kind().unwrap();
+                    if module.is_export() {
+                        return true;
+                    }
                 }
                 AstKind::ExportNamedDeclaration(_) | AstKind::ExportDefaultDeclaration(_) => {
                     return true;
```

Comment:
There is a duplicate branch, `ModuleDeclaration` has covered `AstKind::ExportNamedDeclaration(_)` and `AstKind::ExportDefaultDeclaration(_)`. 
In addition, I've looked into the code here, this is checks whether a symbol has been exported, and only `ExportNamedDeclaration` and `ExportDefaultDeclaration` can produce a Symbol, so the `module.is_export()` is unnecessary.

Reply:
I tried removing the `is_export`, but it caused some tests to fail, so I've left it in for now.

Reply:
Ok, I will take a further look at this.